packages/url_launcher/url_launcher_ios/lib/url_launcher_ios.dart

```diff
@@ -45,11 +45,79 @@ class UrlLauncherIOS extends UrlLauncherPlatform {
     required bool universalLinksOnly,
     required Map<String, String> headers,
     String? webOnlyWindowName,
-  }) {
+  }) async {
+    final PreferredLaunchMode mode;
     if (useSafariVC) {
+      mode = PreferredLaunchMode.inAppBrowserView;
+    } else if (universalLinksOnly) {
+      mode = PreferredLaunchMode.externalNonBrowserApplication;
+    } else {
+      mode = PreferredLaunchMode.externalApplication;
+    }
+    return launchUrl(
+        url,
+        LaunchOptions(
+            mode: mode,
+            webViewConfiguration: InAppWebViewConfiguration(
+                enableDomStorage: enableDomStorage,
+                enableJavaScript: enableJavaScript,
+                headers: headers)));
+  }
+
+  @override
+  Future<bool> launchUrl(String url, LaunchOptions options) async {
```

Comment:
Technically this isn't required for this PR, but it seemed like a good opportunity to add `launchUrl` since it's very similar to the Android version. I can make this a separate PR later if preferred.